clients/gtest/rocblas_gtest_main.cpp

```diff
@@ -188,17 +209,17 @@ extern "C" void rocblas_test_signal_handler(int sig)
     if(!pthread_equal(pthread_self(), rocblas_test_sighandler_tid))
     {
         pthread_kill(rocblas_test_sighandler_tid, sig);
-        sleep(1);
-        errno = e; // Restore errno
+        errno = saved_errno;
         return;
     }
 
-    // Jump back to the handler code
+    errno = saved_errno;
+
+    // Jump back to the handler code after setting signal_received
     // Note: This bypasses stack unwinding, and may lead to memory leaks, but
-    // it is better than crashing. Throwing exceptions from signal handlers is
-    // poorly supported, and may result in recursive calls to std::terminate.
-    errno = e; // Restore errno
-    siglongjmp(rocblas_sigjmp_buf, sig);
+    // it is better than crashing.
+    signal_received = sig;
```

Comment:
We would have to list every signal in a `switch` statement if we send the signal through `siglongjmp`, so we set a global variable instead.